app/react/docker/stacks/stack.service.ts

```diff
@@ -0,0 +1,25 @@
+import { useQuery } from 'react-query';
+
+import axios from '@/portainer/services/axios';
+import { withError } from '@/react-tools/react-query';
+
+import { StackFile, StackId } from './types';
+
+const queryKeys = {
+  stackFile: (stackId?: StackId) => ['stacks', stackId, 'file'],
+};
+
+export function useStackFile(stackId?: StackId) {
+  return useQuery(queryKeys.stackFile(stackId), () => getStackFile(stackId), {
+    ...withError('Unable to retrieve stack'),
+    enabled: !!stackId,
+  });
+}
+
+async function getStackFile(stackId?: StackId) {
+  if (!stackId) {
+    return Promise.resolve(undefined);
+  }
+  const { data } = await axios.get<StackFile>(`/stacks/${stackId}/file`);
+  return data;
+}
```

Comment:
Your right, stacks are for both Kube and Docker, I'll move it if it's not a big refactor

Reply:
I moved most of these files into /react/portainer/stacks because they're shared by both Docker and Kubernetes environments. There was a docker only file that I kept in the same directory.